packages/request-client.js/src/api/payment-network/erc20/proxy-info-retriever.ts

```diff
@@ -0,0 +1,117 @@
+import { ethers } from 'ethers';
+import * as Types from '../../../types';
+
+// The ERC20 proxy smart contract ABI fragment containing TransferWithReference event
+const erc20proxyContractAbiFragment = [
+  {
+    anonymous: false,
```

Comment:
I would suggest we start using the string version of ABIs.
I find something like `"function balanceOf(address _owner) view returns (uint256 balance)"` (random example) a lot more readable than this.